addons/web_editor/static/src/js/common/grid_layout_utils.js

```diff
@@ -83,16 +83,16 @@ export function _gridCleanUp(rowEl, columnEl) {
  * @param {Element} containerEl element with the class "container"
  */
 export function _toggleGridMode(containerEl) {
-    let rowEl = containerEl.querySelector('.row');
-    // For the snippets having text outside of the row (and therefore not in a
-    // column), create a column and put the text in it so it can also be placed
-    // in the grid.
-    const textEls = [...containerEl.children].filter(el => el.nodeName !== 'DIV');
-    if (rowEl && textEls.length > 0) {
+    let rowEl = containerEl.querySelector(':scope > .row');
+    // For the snippets having elements outside of the row (and therefore not in
+    // a column), create a column and put these elements in it so they can also
+    // be placed in the grid.
+    const outOfRowEls = [...containerEl.children].filter(el => !el.classList.contains('row'));
+    if (rowEl && outOfRowEls.length > 0) {
         const columnEl = document.createElement('div');
         columnEl.classList.add('col-lg-12');
-        for (let i = textEls.length - 1; i >= 0; i--) {
-            columnEl.prepend(textEls[i]);
+        for (let i = outOfRowEls.length - 1; i >= 0; i--) {
+            columnEl.prepend(outOfRowEls[i]);
```

Comment:
What if we have a structure like this
```html
<div class="container">
        <div class="row">
                <div class="col-lg-4" ></div>
                <div class="col-lg-4" ></div>
                <div class="col-lg-4" ></div>
        </div>
        <div class="row">
                <div class="col-lg-4" ></div>
                <div class="col-lg-4" ></div>
                <div class="col-lg-4" ></div>
        </div>
</div>
```
One container with two `.row`... The second one will be skipped no ?

Reply:
Ok let's say it is a limitation of the current options.